File: src/quartzite_test/MyDynamicClassLoader.clj
(ns quartzite-test.MyDynamicClassLoader
  (:gen-class
    :extends clojure.lang.DynamicClassLoader
    :implements [org.quartz.spi.ClassLoadHelper]
    :exposes-methods {loadClass parentLoadClass}))

(defn -initialize [this]
  nil)

(defn load-class [loader name]
  "This is named with a dashified format to avoid a StackOverflowError by
   endlessly calling loadClass of _this_ class and not the parent."
  (.parentLoadClass loader name true))

(defn -loadClass
  ([this name]
   (load-class this name))
  ([this name _]
   (load-class this name)))

(defn -getClassLoader [this]
  this)

File: src/java/quartzitetest/MyDynamicClassLoader.java
package quartzitetest;

/**
 * Created by danielle on 2/6/17.
 */
import clojure.lang.DynamicClassLoader;
import org.quartz.spi.ClassLoadHelper;

public class MyDynamicClassLoader extends DynamicClassLoader implements ClassLoadHelper {
    public void initialize() {}

    @SuppressWarnings("unchecked")
    public <T> Class<? extends T> loadClass(String name, Class<T> clazz)
            throws ClassNotFoundException {
        return (Class<? extends T>) loadClass(name);
    }

    public ClassLoader getClassLoader() {
        return this;
    }
}